Enhances the counts command to provide better wallet analysis and visualization:

vexr opened the pull request:
- Sort wallets by transfer count and add percentage of total XDM transfers for each wallet
  - Add block range metadata (consensus/domain start/end heights) to JSON output
  - Convert output from objects to sorted arrays for improved readability
  - Add `counts:html` command to generate interactive HTML leaderboard with wallet search
  - Improve .env loading to support both root and local configurations
  
<img width="1435" height="841" alt="Screenshot From 2025-10-02 04-54-52" src="https://github.com/user-attachments/assets/72cc8421-b964-4519-acad-ca100e2c484d" />

vexr committed:
Enhance counts output with percentages, block ranges, and optional HTML leaderboard

0xjeremyfrank commented on apps/crossing-the-narrow-sea/src/counts.ts:
```diff
@@ -62,6 +85,227 @@ const writeJsonFile = (filePath: string, data: unknown): void => {
   fs.writeFileSync(filePath, JSON.stringify(data, null, 2) + '\n', 'utf8')
 }
 
+const writeHtmlFile = (filePath: string, result: CountsResult): void => {
```
To keep this file cleaner, let's move the html generating code to its own file.

0xjeremyfrank reviewed (changes requested):
Looks great, I just have one nit.

vexr committed:
Extract HTML leaderboard generation to dedicated module

vexr commented:
### As requested:
- [x] Refactored HTML generation into dedicated `htmlLeaderboard.ts` module for better separation of concerns  
- [x] Extracted shared type definitions to `types.ts` for improved code reusability and type safety  
- [x] Increased wallet search field width from 500px to 600px for better address visibility

0xjeremyfrank reviewed (approved):
LGTM! Thank you for your contribution @vexr!

jim-counter reviewed (approved):
LGTM!
